src/bson.c

```diff
@@ -1101,6 +1101,9 @@ static void phongo_bson_append(bson_t *bson, php_phongo_bson_flags_t flags, cons
 		case IS_REFERENCE:
 			ZVAL_DEREF(entry);
 			goto try_again;
+		case IS_UNDEF:
+        		  bson_append_undefined(bson, key, key_len);
```

Comment:
This appends deprecated type 0x06, which correlates with an `undefined` value in JavaScript.

The `IS_UNDEF` constant here indicates an uninitialized or unset zval. To quote [this article](https://nikic.github.io/2015/05/05/Internal-value-representation-in-PHP-7-part-1.html) for PHP 7:

> The `IS_UNDEF` type is used in places where previously a `NULL` zval pointer (not to be confused with an `IS_NULL` zval) was used. For example, in the refcounting examples above the `IS_UNDEF` type is set for variables that have been unset.

@derickr: Based on https://github.com/mongodb/mongo-php-driver/issues/375, I think we should consider adding the field name to the exception message.

@sandipmavani: Can you attempt improving the exception message a few lines below for the `default:` case?

```
phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Got unsupported type %d '%s for field %s'", Z_TYPE_P(entry), zend_get_type_by_const(Z_TYPE_P(entry), key));
```

With that in place, you should have more knowledge of what field in the PHP array/object is uninitialized. I think this is likely a case of memory corruption. Given the field name, you may be able to conclude that it should always have a value in your application, which would further confirm the corruption hypothesis.